### ISSUE TITLE
Should tabs be accepted (and converted to spaces)

### ISSUE THREAD

**Issue by avh4** [avh4/elm-format#185]
elm-compiler does not accept tabs.  It's possible that elm-format could accept tabs and convert them into spaces.

**Comment by sporto** [comment]
Yes please, I was teaching Elm to my kids using https://runelm.io/ and they really struggled with the spaces for indentation. They kept adding tabs when hitting the tab key and the compiler will complain.

Seems they are running elm-format, but it only works if you send valid code (no tabs). It would be great if elm-format would convert the tabs to spaces so this is not an issue.

**Comment by robx** [comment]
I just ran into this (and #662 -- the messages are really quite confusing) today. My naïve expectation would be that elm-format would "simply" convert tabs to spaces, but I now realize it's not that simple because the depth of indentation matters.

I think what I'd like to see is:
1. expand all tabs into spaces automatically (in some simple way, e.g. just expand to 8 spaces regardless) while reading the file
2. if the above code triggered (i.e., there were any tabs outside string literals), add a parenthetical to error output warning that tabs have been converted which might affect things unexpectedly):
```
<error message...>

Note: I ran into tab characters while parsing your code and converted them to spaces automatically.
```
